yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnShuffleEncryptionSuite.scala

```diff
@@ -0,0 +1,307 @@
+        tmp.asInstanceOf[File].renameTo(mergedOutputFile)
+      }
+    }).when(blockResolver).writeIndexFileAndCommit(anyInt(), anyInt(), any(classOf[Array[Long]]),
+          any(classOf[File]))
+
+    when(diskBlockManager.createTempShuffleBlock()).thenAnswer(
+      new Answer[(TempShuffleBlockId, File)]() {
+        override def answer(invocationOnMock: InvocationOnMock): (TempShuffleBlockId, File) = {
+          val blockId = new TempShuffleBlockId(UUID.randomUUID())
+          val file = File.createTempFile("spillFile", ".spill", tempDir)
+          spillFilesCreated.add(file)
+          (blockId, file)
+        }
+      })
+
+    when(taskContext.internalMetricsToAccumulators).thenReturn(null)
+    when(taskContext.taskMetrics()).thenReturn(taskMetrics)
+    when(shuffleDep.serializer).thenReturn(Option.apply(serializer))
+    when(shuffleDep.partitioner).thenReturn(hashPartitioner)
+    when(taskContext.taskMetrics()).thenReturn(taskMetrics)
+  }
+
+  private[this] def initialMocksForReader(): Unit = {
+    // Create a return function to use for the mocked wrapForCompression method to initial a
+    // compressed input stream if spark.shuffle.compress is enabled
+    val compressionFunction = new Answer[InputStream] {
+      override def answer(invocation: InvocationOnMock): InputStream =
+        if (conf.getBoolean("spark.shuffle.compress", false)) {
+          CompressionCodec.createCodec(conf).compressedInputStream(
+            invocation.getArguments()(1).asInstanceOf[InputStream])
+        } else {
+          invocation.getArguments()(1).asInstanceOf[InputStream]
+        }
+    }
+
+    // Setup the mocked BlockManager to return RecordingManagedBuffers.
+    val localBlockManagerId = BlockManagerId("test-client", "test-client", 1)
+    when(blockManager.blockManagerId).thenReturn(localBlockManagerId)
+
+    var startOffset = 0L
+    for (mapId <- 0 until NUM_PARTITITONS) {
+      val bytes = new Array[Byte](partitionSizesInMergedFile(mapId).toInt)
+      val in = new FileInputStream(mergedOutputFile)
+      try {
+        ByteStreams.skipFully(in, startOffset)
+        in.read(bytes)
+      } finally {
+        in.close()
+      }
+      // Create a ManagedBuffer with the shuffle data.
+      val nioBuffer = new NioManagedBuffer(ByteBuffer.wrap(bytes))
+      val managedBuffer = new RecordingManagedBuffer(nioBuffer)
+      startOffset += partitionSizesInMergedFile(mapId)
+      // Setup the blockManager mock so the buffer gets returned when the shuffle code tries to
+      // fetch shuffle data.
+      val shuffleBlockId = ShuffleBlockId(SHUFFLE_ID, mapId, REDUCE_ID)
+      when(blockManager.getBlockData(shuffleBlockId)).thenReturn(managedBuffer)
+      when(blockManager.wrapForCompression(meq(shuffleBlockId), isA(classOf[InputStream])))
+          .thenAnswer(compressionFunction)
+    }
+
+    when(mapOutputTracker.getMapSizesByExecutorId(SHUFFLE_ID, REDUCE_ID, REDUCE_ID + 1)).thenReturn
+    {
```

Comment:
nit: this syntax looks weird. Can you store the result of executing the block in a variable and call `.thenReturn(result)`?